tests/views/test_learner_metrics_viewset.py

```diff
@@ -0,0 +1,156 @@
+            "enrollments": [
+                {
+                    "id": 9,
+                    "course_id": "course-v1:StarFleetAcademy+SFA01+2161",
+                    "date_enrolled": "2020-02-24",
+                    "is_enrolled": true,
+                    "progress_percent": 1.0,
+                    "progress_details": {
+                        "sections_worked": 20,
+                        "points_possible": 100.0,
+                        "sections_possible": 20,
+                        "points_earned": 50.0
+                    }
+                }
+            ]
+        }
+    ```
+    """
+    base_request_path = 'api/learner-metrics/'
+    view_class = LearnerMetricsViewSet
+
+    @pytest.fixture(autouse=True)
+    def setup(self, db, settings):
+        if organizations_support_sites():
+            settings.FEATURES['FIGURES_IS_MULTISITE'] = True
+        super(TestLearnerMetricsViewSet, self).setup(db)
+
+    def make_caller(self, site, users):
+        """Convenience method to create the API caller user
+        """
+        if organizations_support_sites():
+            # TODO: set is_staff to False after we have test coverage
+            caller = UserFactory(is_staff=True)
+            map_users_to_org_site(caller=caller, site=site, users=users)
+        else:
+            caller = UserFactory(is_staff=True)
+        return caller
+
+    def make_request(self, monkeypatch, request_path, site, caller, action):
+        """Convenience method to make the API request
+
+        Returns the response object
+        """
+        request = APIRequestFactory().get(request_path)
+        request.META['HTTP_HOST'] = site.domain
+        monkeypatch.setattr(django.contrib.sites.shortcuts,
+                            'get_current_site',
+                            lambda req: site)
+        force_authenticate(request, user=caller)
+        view = self.view_class.as_view({'get': action})
+        return view(request)
+
+    def test_list_method_all(self, monkeypatch, enrollment_test_data):
+        """INCOMPLETE TEST
+        """
+        site = enrollment_test_data['site']
+        users = enrollment_test_data['users']
+        enrollments = enrollment_test_data['enrollments']
+
+        caller = self.make_caller(site, users)
+        other_site = SiteFactory()
+        assert site.domain != other_site.domain
+
+        response = self.make_request(request_path=self.base_request_path,
+                                     monkeypatch=monkeypatch,
+                                     site=site,
+                                     caller=caller,
+                                     action='list')
+
+        assert response.status_code == status.HTTP_200_OK
+        assert is_response_paginated(response.data)
+        results = response.data['results']
+        # Check user ids
+        result_ids = [obj['id'] for obj in results]
+        assert set(result_ids) == set([obj.user_id for obj in enrollments]+[caller.id])
```

Comment:
This is what is failing because the result_ids is also including users created in the base test class from which this class inherits

Reply:
This is fixed